src/coreComponents/linearAlgebra/interfaces/direct/SuiteSparse.hpp

```diff
@@ -20,285 +20,133 @@
-   * @return info error code
-   */
-  int solveWorkingRank( real64 * b, real64 * x, bool transpose = false );
-
-  /**
-   * @brief Sycronizes times across ranks
-   */
-  void syncTimes();
-
-  /**
-   * @brief Estimates the condition number of the matrix
-   * @return the estimated condition number
-   */
-  real64 condEst() const;
-
-  /**
-   * @brief Estimates the relative tolerance for the matrix
-   * @return the relative tolerance (condEst * eps)
-   */
-  real64 relativeTolerance() const;
-
-  /**
-   * @brief Deallocates a SuiteSparse data structure
-   */
-  void destroy();
-
-  /**
-   * @brief Sets the working rank
-   * @param[in] workingRank the working rank
-   */
-  void setWorkingRank( int const workingRank );
+  virtual ~SuiteSparse() override;
 
-  /**
-   * @brief Returns the working rank
-   * @return the working rank
-   */
-  int workingRank() const;
-
-  /**
-   * @brief Sets the working rank in the sub-communicator
-   * @param[in] subCommWorkingRank the working rank in the sub-communicator
-   */
-  void setSubCommWorkingRank( int const subCommWorkingRank );
-
-  /**
-   * @brief Returns the working rank in the sub-communicator
-   * @return the working rank in the sub-communicator
-   */
-  int subCommWorkingRank() const;
-
-  /**
-   * @brief Sets the communicator
-   * @param[in] comm the MPI communicator
-   */
-  void setComm( MPI_Comm const comm );
+  using Base::ready;
+  using Base::matrix;
 
   /**
-   * @brief Returns the communicator
-   * @return the communicator
+   * @brief Compute the preconditioner from a matrix.
+   * @param mat the matrix to precondition.
    */
-  MPI_Comm getComm() const;
+  virtual void setup( Matrix const & mat ) override;
 
   /**
-   * @brief Sets the subcommunicator
-   * @param[in] subComm the MPI subcommunicator
+   * @brief Apply operator to a vector.
+   * @param src input vector (x)
+   * @param dst output vector (b)
+   *
+   * @warning @p src and @p dst cannot alias the same vector.
    */
-  void setSubComm( MPI_Comm const subComm );
+  virtual void apply( Vector const & src, Vector & dst ) const override;
 
   /**
-   * @brief Returns the subcommunicator
-   * @return the subcommunicator
+   * @brief Apply transpose operator to a vector.
+   * @param src input vector (x)
+   * @param dst output vector (b)
+   *
+   * @warning @p src and @p dst cannot alias the same vector (some implementations may allow this).
    */
-  MPI_Comm getSubComm() const;
+  void applyTranspose( Vector const & src, Vector & dst ) const;
 
   /**
-   * @brief Returns the number of rows
-   * @returns the number of rows
+   * @brief Clean up the preconditioner setup.
    */
-  SSInt numRows() const;
+  virtual void clear() override;
 
   /**
-   * @brief Returns the number of columns
-   * @return the number of columns
+   * @brief Solve the system with a particular rhs
+   * @param [in] rhs system right hand side
+   * @param [out] sol system solution
```

Comment:
Indeed, it's a mix of styles depending on who wrote the initial documentation (and things tend to get copy-pasted too).
Personally, I feel this is often not needed, because the function signature (i.e. `const`-ness of parameters) expresses the intent correctly in most cases.Consistency would be great though, so maybe we can reach a decision during next Doc Day and apply the policy.

Reply:
Actually, I really hope C++ will soon allow for declaring intent directly ([Herb Sutter - CppCon 2020](https://youtu.be/6lurOCdaj0Y?t=919)).

Reply:
Yeah, this is a good proposal, but I feel it's too radical for many standards committee members :)